### ISSUE TITLE
Unable to use custom type declarations in VSCode

### ISSUE THREAD

**Issue by jonathansayag** [SqrTT/prophet#330]
**Describe the bug**
When Prophet is enabled in VSCode, custom types definitions are not visible. As i know from https://github.com/SqrTT/prophet/issues/276, Prophet comes with its own custom type definitions & disable built in TS language server.

**To Reproduce**
1. Create a custom file at the root of the project in `types/my/declaration/MyClass.d.ts` : 

  ```
  export function myFunction(myParam1: string): string;
  export function myFunction2(myParam2: string): string;
  ```

2. Create a `jsconfig.json` at the root of the project : 

  ```
  {
      "compilerOptions": {
          "noLib": true,
          "target": "es5",
          "baseUrl": "./",
          "paths": {
              "*" : ["./*", "modules/*", "../types/*"]
          }
      },
      "typeAcquisition": {
          "enable": false
      },
      "include": [
          "../types/*.d.ts",
          "../types/**/*.d.ts",
          "./app_custom_cartridge/**/*.js"
      ]
  }
  ```

3. In my file `app_custom_cartridge/cartridge/scripts/test-script.js`, i'm trying to import type declaration : 

```
/**
 *
 * @param {import('my/declaration/MyClass')} myClass
 * @return {string}
 */
var myTestFunction = function(myClass) {
    myClass.myFunction2(); // <---- auto-complete not working here
};
```

**Expected behavior**
VSCode detect the `my/declaration/MyClass` import & intellisense can propose me the functions `myFunction` or `myFunction2`.

**Actual behaviour**
VSCode shows me : 
```
Cannot find module 'my/declaration/MyClass' or its corresponding type declarations.(prophet-ts(2307))
```

**Additional context**
Note that if i disable Prophet, everything works & Vscode is enable to go to my custom type declaration.

**Comment by SqrTT** [comment]
Hi @jonathansayag 
Thanks for reporting.
> Prophet comes with its own custom type definitions & disable built in TS language server.

That's correct. Prophet disable built in TS language server for SFCC server side code and provide it's own assistance within context of SFCC. Basically Prophet assistance is modified TypeScript Server with knowledge of SFCC. Modified TypeScript Server doesn't take into account `jsconfig.json` since there are nothing to configure for SFCC.

Regarding your example with `my/declaration/MyClass`: Modified TypeScript Server also have changed modules resolution strategy to fit SFCC logic (resolving paths like `~/someFolder/file.js`, `*/someFolder/etc.js`, `module.superModule` so on). So in your example Prophet tries to find cartridge `me` and fails (just thing of it like it's executed on server side)

Basically you may place definitions along `.js` file to be resolved correctly or use JSDoc to define types within file itself

**Comment by jonathansayag** [comment]
Hi @SqrTT, thanks for your reply.

As expected, it is working by using the cartridge path.

Thank you for your help !
